### PR TITLE
Labeled/Segmented Point Cloud Resource Type

### PR DESCRIPTION
**Pull request by mfehr**
Adds a new point cloud resource type, with XYZLabel and includes the corresponding conversion functions between `sensor_msgs::PointCloud2`, `pcl::PointCloud<pcl::PointXYZL>` and our own `resources::PointCloud` type. The idea is to use these resources to attach LOAM "feature" point clouds or semantic segmentations from input like voxblox++.

TODO:
 - Test in live system with LOAM point clouds

### CODE REVIEW

File: backend/map-resources/include/map-resources/resource-conversion-inl.h [hunk 1]
```diff
@@ -279,7 +314,7 @@ void getScalarFromPointCloud(
     const PointCloudType& /*point_cloud*/, const size_t /*index*/,
     float* /*scalar*/) {
   LOG(FATAL) << "This point cloud either does not support scalars/intesities "
-                "or it is not "
```
Comment (floriantschopp) [inline backend/map-resources/include/map-resources/resource-conversion-inl.h:282]: weird formatting, should be the same line i guess.

File: backend/map-resources/include/map-resources/resource-conversion-inl.h [hunk 2]
```diff
@@ -303,6 +338,27 @@ void getScalarFromPointCloud(
     const pcl::PointCloud<pcl::OusterPointType>& point_cloud,
     const size_t index, float* scalar);
 
+template <typename PointCloudType>
+void getLabelFromPointCloud(
+    const PointCloudType& /*point_cloud*/, const uint32_t /*label*/,
+    float* /*scalar*/) {
+  LOG(FATAL) << "This point cloud either does not support labels"
+             << "or it is not "
```
Comment (floriantschopp) [inline backend/map-resources/include/map-resources/resource-conversion-inl.h:346]: same as above

File: backend/map-resources/include/map-resources/resource-conversion-inl.h [hunk 3]
```diff
@@ -303,6 +338,27 @@ void getScalarFromPointCloud(
...
+             << "implemented!";
```
Comment (mfehr) [inline backend/map-resources/include/map-resources/resource-conversion-inl.h:347]:
```suggestion
  LOG(FATAL) << "This point cloud either does not support labels"
             << "or it is not implemented!";
```

**Comment by mfehr** [comment]
test this please

**Pull request merged**